openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+	PChain string          `json:"pchain,omitempty"` // Payment ID chain string containing embedded syntax described in the TAG Payment ID Protocol. NOTE that the authentication features in this Source object combined with the “ads.txt” specification may lead to the future deprecation of this attribute.
+	Ext    json.RawMessage `json:"ext,omitempty"`    //	Optional exchange-specific extensions.
+}
+
+// Item object represents a unit of goods being offered for sale either on the open market or in relation to a private marketplace deal.
+// The id attribute is required since there may be multiple items being offered in the same bid request and bids must reference the specific item of interest.
+// This object interfaces to Layer-4 domain objects for deeper specification of the item being offered (e.g., an impression).
+type Item struct {
+	ID      string          `json:"id"`                // Required. A unique identifier for this item within the context of the offer (typically starts with 1 and increments).
+	Qty     int             `json:"qty,omitempty"`     // Default: 1. The number of instances (i.e., "quantity") of this item being offered (e.g., multiple identical impressions in a digital out-of-home scenario).
+	Seq     *int            `json:"seq,omitempty"`     // If multiple items are offered in the same bid request, the sequence number allows for the coordinated delivery.
+	Flr     float64         `json:"flr,omitempty"`     // Minimum bid price for this item expressed in CPM.
+	FlrCur  string          `json:"flrcur,omitempty"`  // Default: "USD". Currency of the “flr” attribute specified using ISO-4217 alpha codes.
+	Exp     int64           `json:"exp,omitempty"`     // Advisory as to the number of seconds that may elapse between auction and fulfilment.
+	DT      string          `json:"dt,omitempty"`      // Date/time when the item is expected to be fulfilled (e.g. when a DOOH impression will be displayed) using the W3C-defined ISO-8601 format.
+	Dlvy    int8            `json:"dlvy,omitempty"`    // Default: 0. Item (e.g., an Ad object) delivery method required, where 0 = either method, 1 = the item must be sent as part of the transaction (e.g., by value in the bid itself, fetched by URL included in the bid), and 2 = an item previously uploaded to the exchange must be referenced by its ID. Note that if an exchange does not supported prior upload, then the default of 0 is effectively the same as 1 since there can be no items to reference.
+	Metric  []Metric        `json:"metric,omitempty"`  // An array of “Metric” objects.
+	Deal    []Deal          `json:"deal,omitempty"`    // Array of “Deal” objects that convey special terms applicable to this item.
+	Private Bool            `json:"private,omitempty"` // Default: 0 (false). Indicator of auction eligibility to seats named in “Deal” objects, where 0 (false) = all bids are accepted, 1 (true) = bids are restricted to the deals specified and the terms thereof.
```

Comment:
Candidate to become a pointer (already bumped into it when writing marshalling tests)